examples/viewport.c

```diff
@@ -34,12 +34,12 @@ main(int argc, char *argv[])
     }
 
     g_set_prgname("view-stack");
-    cog_modules_add_directory(g_getenv("COG_MODULEDIR") ?: COG_MODULEDIR);
 
-    g_autoptr(GError)      error = NULL;
-    g_autoptr(CogShell)    shell = cog_shell_new(g_get_prgname(), FALSE);
-    g_autoptr(CogPlatform) platform = cog_platform_configure(NULL, NULL, "COG", shell, &error);
-    if (!platform)
+    g_autoptr(CogShell) shell = cog_shell_new(g_get_prgname(), FALSE);
```

Comment:
I think the platform has to be created first. Even before than the  shell. The shell needs a already created platform. For example, in case of the shell initializes a default viewport containing a default view.

Reply:
That is not an issue, because the shell uses `cog_platform_get()`, and that will create the platform instance when needed.

The only case in which the platform instance needs to be created first is to manually specify which one to use with the API, and we have `cog_init()` for that.

Reply:
Using `cog_init("wl", NULL);`

Reply:
https://github.com/Igalia/cog/pull/659 makes this even more robust and allows to dont need to use the `cog_init()`.